core-bundle/src/Event/FrontendMenuEvent.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Event;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class FrontendMenuEvent extends Event
+{
+    private FactoryInterface $factory;
+    private ItemInterface $menu;
+    private int $pid;
+    private array $options;
+
+    public function __construct(FactoryInterface $factory, ItemInterface $menu, int $pid, array $options)
+    {
+        $this->factory = $factory;
```

Comment:
The `Contao\CoreBundle\Event\MenuEvent` does the same, but I am also not sure why.

Reply:
Official docs do so as well: https://symfony.com/bundles/KnpMenuBundle/current/events.html#create-the-event-object
It may be because the concrete Factory can differ from the one in the container, no?.

Reply:
Or because KnpMenu also works without a DI container. But I guess its fine this way as well 😊